opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,334 @@
+
+    Cumulative values can go up or stay the same, but can never go down.
+    Cumulative values cannot be negative.
+    """
+
+    def get_or_create_time_series(self,
+                                  label_values: LabelValues
+                                  ) -> 'CounterTimeSeries':
+        """Gets a `CounterTimeSeries` with a cumulative int value."""
+
+
+class GaugeFloat(Metric):
+    """A gauge type metric that holds float values.
+
+    Cumulative value can go both up and down. Values can be negative.
+    """
+
+    def get_or_create_time_series(self,
+                                  label_values: LabelValues
+                                  ) -> 'GaugeTimeSeries':
+        """Gets a `GaugeTimeSeries` with a cumulative float value."""
+
+
+class GaugeInt(Metric):
+    """A gauge type metric that holds int values.
+
+    Cumulative value can go both up and down. Values can be negative.
+    """
+
+    def get_or_create_time_series(self,
+                                  label_values: LabelValues
+                                  ) -> 'GaugeTimeSeries':
+        """Gets a `GaugeTimeSeries` with a cumulative int value."""
+
+
+class MeasureFloat(Metric):
+    """A measure type metric that holds float values.
+
+    Measure metrics represent raw statistics that are recorded.
+    """
+
+    def get_or_create_time_series(self,
+                                  label_values: LabelValues
+                                  ) -> 'MeasureTimeSeries':
+        """Gets a `MeasureTimeSeries` with a cumulated float value."""
+
+
+class MeasureInt(Metric):
+    """A measure type metric that holds int values.
+
+    Measure metrics represent raw statistics that are recorded.
+    """
+
+    def get_or_create_time_series(self,
+                                  label_values: LabelValues
+                                  ) -> 'MeasureTimeSeries':
+        """Gets a `MeasureTimeSeries` with a cumulated int value."""
+
+
+class MeasureBatch:
+
+    def record(self,
+               metrics: List['Metric'],
+               values: List[Union[float, int]]) -> None:
+        """Records multiple observed values simultaneously.
+
+        Args:
+            metric: A list containing the `Metric` s to be recorded
+            values: A list containing the values to record
+        """
+
+
+class LabelKey:
+    """The label keys associated with the metric.
+
+    :type key: str
+    :param key: the key for the label
+
+    :type description: str
+    :param description: description of the label
+    """
+    def __init__(self,
+                 key: str,
+                 description: str) -> None:
+        self.key = key
+        self.description = description
+
+
+class LabelValue:
+    """The label values associated with a TimeSeries.
+
+    :type value: str
+    :param value: the value for the label
+    """
+    def __init__(self,
+                 value: str) -> None:
+        self.value = value
```

Comment:
I wonder if there's any good reasons to have mandatory containers for label values? Currently it seems to be engineered like that just for the sake of completeness.

As far as I understand majority of industry solutions, there's no support for anything but strings for labels, and it's unlikely to change in the years to come.

Reply:
This is good point and a common ask. I think it will make sense to simply use strings for the label values.

Reply:
I think each language should decide how to handle this question. If there is some kind of built-in support for key-value in the language, they should prefer that. In the Go repository, we have a common `core.KeyValue` type that can be used as both an event label and a metric label. In logging (events), it's more common to have non-string values. 

I'm not sure the API needs to specify anything about how a vendor should behave when, say, a numeric value is passed as the value for a label (and the compiler didn't prevent it). I would say the default behavior should coerce values to a string silently, not something to worry about.